src/Microsoft.AspNetCore.Razor.Language/RazorConfiguration.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright(c) .NET Foundation.All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Razor.Language
+{
+    public sealed class RazorConfiguration
+    {
+        public static readonly RazorConfiguration DefaultRuntime = new RazorConfiguration(RazorLanguageVersion.Latest, designTime: false);
+        public static readonly RazorConfiguration DefaultDesignTime = new RazorConfiguration(RazorLanguageVersion.Latest, designTime: true);
```

Comment:
I'm really not sure this is a thing. I don't know exactly what a default `RazorConfiguration` means

Reply:
Good point, in RazorEngine land it makes sense because we have separate Runtime/DesignTIme methods that are used to create goodness but in the new RazorProjectEngine world you'll have to go out of your way to have a design time configuration; at which point you might as well construct the configuration from the ground up. I'll address this in the coming PR.

Reply:
Going to leave a runtime default as `RazorConfiguration.Default` as a convenience for tests.

Reply:
OK, but even for runtime I'm not sure that we'll ever have a **GLOBAL DEFAULT CONFIGURATION**

Reply:
> OK, but even for runtime I'm not sure that we'll ever have a GLOBAL DEFAULT CONFIGURATION

Why wouldn't RazorLanguageVersion.Latest + designTime: false not be a default config?

Reply:
Because what code does it generate?

Reply:
Myabe I'm geting too far inside my own head here.

Reply:
> Because what code does it generate?

Code compliant with the latest Razor version. I think you and I might see different futures for the `RazorConfiguration` class. I don't envision `RazorConfiguration` being the super config class that also encompasses Razor extensibility. I view a higher layer containing that logic (like we have today in `ProjectExtensibilityConfiguration`

Reply:
Ok, let's go with your idea for now then. We can revisit - I think it will be easy to figure out when we've written more codez.